src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ContigAligner implements Closeable {
+
+    final BwaIndex index;
+    final BwaMem bwaMem;
+
+    public ContigAligner(final File referenceFile) {
+        BWANativeLibrary.load();
+        try {
+            index = new BwaIndex(referenceFile);
+        } catch (IOException e) {
+            throw new GATKException("Unable to load reference", e);
+        }
+        bwaMem = new BwaMem(index);
+
+    }
+
+    public List<AssembledBreakpoint> alignContigs (final RunSGAViaProcessBuilderOnSpark.ContigsCollection contigsCollection) {
+        final List<AssembledBreakpoint> assembledBreakpoints = new ArrayList<>();
+
+
+            try {
+                for(final Tuple2<RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID, RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigSequence> contigInfo : contigsCollection.getContents()) {
+                    final String contigId = contigInfo._1.toString();
+                    final byte[] sequence = contigInfo._2.toString().getBytes();
+                    final AlnRgn[] alnRgns = bwaAlignSequence(bwaMem, contigId, sequence);
+
+                    // todo: parse cigar for internal indels
+                    if (alnRgns.length > 1) {
+
+                        // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                        final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                                .filter(a -> a.getSecondary() == -1)
+                                .map(AlignmentRegion::new)
+                                .sorted(Comparator.comparing(a -> a.startInContig))
+                                .collect(Collectors.toList());
+
+                        AlignmentRegion current = alignmentRegionList.get(0);
+                        AlignmentRegion previous;
+                        for (int i = 0; i < alignmentRegionList.size(); i++) {
+                            if (i > 0) {
+                                previous = current;
+                                current = alignmentRegionList.get(i);
+                                final AssembledBreakpoint assembledBreakpoint = new AssembledBreakpoint();
+                                assembledBreakpoint.contigId = contigId;
+
+                                assembledBreakpoint.region1 = previous;
+                                assembledBreakpoint.region2 = current;
+
+                                assembledBreakpoint.homology = "NA";
+                                if (assembledBreakpoint.region1.endInContig >= assembledBreakpoint.region2.startInContig) {
+                                    assembledBreakpoint.homology = getSequence(contigsCollection, contigId, assembledBreakpoint.region2.startInContig - 1, assembledBreakpoint.region1.endInContig);
```

Comment:
We already have the sequence (`contigInfo._2`).  We can pick off the right stuff with a String substring rather than searching the whole collection for the contig under consideration.